e2e/test/scenarios/joins/reproductions/23293-drill-through-implicit-join-add-remove-column.cy.spec.js

```diff
@@ -0,0 +1,96 @@
+import {
+  restore,
+  popover,
+  openOrdersTable,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS, PRODUCTS } = SAMPLE_DATABASE;
+
+describe("issue 23293", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should retain the filter when drilling through the dashboard card with implicitly added column (metabase#23293)", () => {
+    openOrdersTable();
+
+    cy.findByTestId("viz-settings-button").click();
+    modifyColumn("Product ID", "remove");
+    modifyColumn("Category", "add");
+    cy.wait("@dataset");
+
+    saveQuestion().then(
+      ({
+        response: {
+          body: { id },
+        },
+      }) => {
+        const questionDetails = {
+          query: {
+            "source-table": `card__${id}`,
+            aggregation: [["count"]],
+            breakout: [
+              [
+                "field",
+                PRODUCTS.CATEGORY,
+                {
+                  "source-field": ORDERS.PRODUCT_ID,
+                },
+              ],
+            ],
+          },
+          display: "bar",
+        };
+
+        cy.createQuestionAndDashboard({ questionDetails }).then(
+          ({ body: { dashboard_id } }) => {
+            visitDashboard(dashboard_id);
+          },
+        );
+
+        cy.get(".bar").first().realClick();
+        popover()
+          .findByText(/^See these/)
+          .click();
+
+        cy.findByTestId("qb-filters-panel").should(
+          "contain",
+          "Product → Category is Doohickey",
+        );
+        cy.findAllByTestId("header-cell")
+          .last()
+          .should("have.text", "Product → Category");
+
+        cy.findAllByRole("grid")
+          .last()
+          .as("tableResults")
+          .should("contain", "Doohickey")
+          .and("not.contain", "Gizmo");
+      },
+    );
+  });
+});
+
+/**
+ * @param {string} columnName
+ * @param {("add"|"remove")} action
+ */
+function modifyColumn(columnName, action) {
+  const icon = action === "add" ? "add" : "eye_outline";
+  const iconSelector = `.Icon-${icon}`;
+  const columnSeletor = `draggable-item-${columnName}`;
+  cy.findByTestId(columnSeletor).find(iconSelector).click();
```

Comment:
Actually, I had attempted to fix some of our Cypress commands like `button` or `icon` to work with previous subjects (aka when chaining commands), so we would be able to do `cy.findByTestId(columnSeletor).icon(icon).click()`

However, with the current implementation, it doesn't take previous subjects into account.

https://github.com/metabase/metabase/blob/2e328a254ca97964e209f354416d4151d7030822/e2e/support/commands/ui/icon.js#L2

And the problem even after taking the previous subjects into account is that it fails a lot of tests that's already using chaining syntax, since the result will be just like you call those commands without subjects anyway e.g. `cy.findByText(xxx).icon(icon)` is equivalent to `cy.icon(icon)`, but the fix would change the behavior of those tests.

I want to raise this point to your awareness, but this isn't something we need to be addressed at the moment.

Reply:
Oh, I am aware.

https://github.com/metabase/metabase/pull/31161